pre_commit/git.py

```diff
@@ -43,6 +43,7 @@ def no_git_env(_env: Mapping[str, str] | None = None) -> dict[str, str]:
             'GIT_SSL_NO_VERIFY', 'GIT_CONFIG_COUNT',
             'GIT_HTTP_PROXY_AUTHMETHOD',
             'GIT_ALLOW_PROTOCOL',
+            'SSH_ASKPASS', 'GIT_ASKPASS', 'GIT_USERNAME', 'GIT_PASSWORD',
```

Comment:
- `SSH_ASKPASS` doesn't start with `GIT_` so it's not relevant
- `GIT_USERNAME` isn't a variable supported by `git`
- `GIT_PASSWORD` isn't a variable supported by `git`

Reply:
oops, good point; removed ssh_askpass. although git_username/git_password isn't directly, it is the default used by jenkins git plugin; so without it theres no way to provide creds down to git from jenkins as a caller without it. is there any harm in adding it ? I don't think you can use jenkins credential store to execute precommit otherwise.

Reply:
jenkins shouldn't be setting environment variables in other tools' namespaces -- I'm fine adding `GIT_ASKPASS` but not the other ones

Reply:
fair enough; removed in latest force push. commit is just GIT_ASKPASS now